ui/src/app/settings/views/Configuration/DeployFormFields/DeployFormFields.js

```diff
@@ -0,0 +1,52 @@
+import PropTypes from "prop-types";
+import React from "react";
+import { useSelector } from "react-redux";
+
+import { extendFormikShape } from "app/settings/proptypes";
+import selectors from "app/settings/selectors";
+import FormikField from "app/base/components/FormikField";
+import Select from "app/base/components/Select";
+
+const DeployFormFields = ({ formikProps }) => {
+  const defaultOSystemOptions = useSelector(
+    selectors.config.defaultOSystemOptions
+  );
+
+  const distroSeriesOptions = useSelector(state =>
+    selectors.general.getOsReleases(state, formikProps.values.default_osystem)
+  );
+
+  return (
+    <>
+      <FormikField
+        label="Default operating system used for deployment"
+        component={Select}
+        options={defaultOSystemOptions}
+        fieldKey="default_osystem"
+        formikProps={formikProps}
+        onChange={e => {
```

Comment:
@squidsoup @huwshimi I'm having an issue here. When changing the value for the top field, the formik value for the second field isn't updated.

I'm trying to do this manually using the built in methods that formik provides but `distroSeriesOptions` doesn't update in time for this to work. I've tried calling `useSelector` within the `onChange` handler but I can't actually do that because of how hooks work.

Reply:
Hmm... I can't reproduce this. When I change the OS the series field does update for me (switches between centos 6/bionic).

Reply:
@huwshimi the second field visibly updates, but the value within formikProps.values itself doesn't. You have to manually click the field for it to update.

Reply:
@steverydz hey steve, did you resolve this issue?

Reply:
@squidsoup yeah, I made a selector that returns all options formatted in a way that you can easily access what you want based on the key